### ISSUE TITLE
Using the dataquality package with setfit

### ISSUE THREAD

**Issue by vahuja4** [huggingface/setfit#356]
As can be seen here, the `dataquality` package is tightly integrated with the `huggingface` package. The `dq` pkg logs certain statistics during the training and sends it to their platform for visual analysis. When I use it with `setfit`, the `dq` pkg fails because it expects the `n_gpu` argument in the `hf trainer` to be set to 1. But it seems that the `n_gpu` argument is not a part of the `setfit` trainer. So, I am not able to use `dq` along with `setfit`. Is there a way to overcome this? Attached is the error snippet
![error](https://user-images.githubusercontent.com/5290075/231991255-4a645cee-1afd-457e-aa12-8ac60152c7cb.jpeg)

**Comment by tomaarsen** [comment]
I'm afraid there's no real way to overcome this. The SetFitTrainer does not subclass the Transformers Trainer, so it doesn't have an `args` parameter and other things that `dataquality` would expect. #265 gets us in that direction, but it still wouldn't use the Transformers Trainer, so it likely still wouldn't work.

**Comment by franz101** [comment]
@vahuja4 I just opened the PR for SetFit support.
#265 is still different architecture than the trainer. I'm sure there is a reasonable way to implement it, either with a custom Trainer Step or specific model architecture on the SetFit side.

For now I hook into the predict_proba function to log the embeddings and logits with [dataquality](https://github.com/rungalileo/dataquality) this works to show prediction errors in [galileo](https://rungalileo.io
<img width="1483" alt="Screenshot 2023-05-03 at 14 22 49" src="https://user-images.githubusercontent.com/18228395/236078626-8c487177-b166-4b3f-8b70-52d8867b1c93.png">
)

**Comment by tomaarsen** [comment]
That is amazing! Well done!
